lib/rubocop/cop/style/string_concatenation.rb

```diff
@@ -15,18 +15,37 @@ module Style
       # lines, this cop does not register an offense; instead,
       # `Style/LineEndConcatenation` will pick up the offense if enabled.
       #
-      # @example
+      # Two modes are supported:
+      # 1. `aggressive` style checks and corrects all occurrences of `+` where
+      # either the left or right side of `+` is a string literal.
+      # 2. `conservative` style on the other hand, checks and corrects only if
+      # left side (receiver of `+` method call) is a string literal.
+      # This is useful when the receiver is some expression that returns string like `Pathname`
```

Comment:
`Pathname.new` returns a `Pathname`, not a string.  `Pathname.new("x") + "y"` also returns a `Pathname`.

I think this comment should read 
> This is useful when the receiver is an object whose `+` operator accepts a string, but does not return a string -- such as `Pathname`